src/classes/species.cpp

```diff
@@ -223,3 +224,68 @@ int Species::nCoordinateSets() const { return coordinateSets_.size(); }
 
 // Return coordinates sets
 const std::vector<std::vector<Vec3<double>>> &Species::coordinateSets() const { return coordinateSets_; }
+
+toml::basic_value<toml::discard_comments, std::map, std::vector> Species::serialize()
+{
+    toml::basic_value<toml::discard_comments, std::map, std::vector> species;
+    if (forcefield_ != nullptr)
+        species["forcefield"] = forcefield_->name().data();
+
+    if (atoms_.size() > 0)
```

Comment:
```suggestion
    if (!atoms_.empty())
```

You'll need this for all the conditions.

The reason for this is that the C++ standard promises that `empty` is always O(1), but `size` can be `O(N)`, depending on the container.  I doubt that we'd move to a container that does that, but it's still good practice to use `empty`.